tilavarauspalvelu/integrations/email/main.py

```diff
@@ -0,0 +1,506 @@
+from .find_recipients import (
+    get_application_email_recipients,
+    get_recipients_for_applications_by_language,
+    get_reservation_email_recipients,
+    get_reservation_staff_notification_recipients,
+)
+from .rendering import render_html, render_text
+from .sending import send_emails_in_batches_task, send_multiple_emails_in_batches_task
+from .template_context import (
+    get_context_for_application_handled,
+    get_context_for_application_in_allocation,
+    get_context_for_application_received,
+    get_context_for_reservation_approved,
+    get_context_for_reservation_cancelled,
+    get_context_for_reservation_confirmed,
+    get_context_for_reservation_modified,
+    get_context_for_reservation_rejected,
+    get_context_for_reservation_requires_handling,
+    get_context_for_reservation_requires_payment,
+    get_context_for_staff_notification_reservation_made,
+    get_context_for_staff_notification_reservation_requires_handling,
+)
+
+if TYPE_CHECKING:
+    from tilavarauspalvelu.models import Reservation
+    from tilavarauspalvelu.typing import Lang
+
+__all__ = [
+    "EmailService",
+]
+
+
+class EmailService:
+    """Service for sending email notifications available in the system."""
+
+    @staticmethod
+    def send_application_received_email(
+        application: Application,
+        *,
+        language: Lang | None = None,
+    ) -> None:
+        """
+        Sends email about the application being received.
+
+        :param application: The application the email concerns.
+        :param language: The language of the email. Determine from application if not given.
+        """
+        # Prevent accidental sending of wrong email.
+        if application.status != ApplicationStatusChoice.RECEIVED:
+            return
+
+        recipients = get_application_email_recipients(application=application)
+        if not recipients:
+            SentryLogger.log_message(
+                "No recipients for application received email",
+                details={"application": application.pk},
+            )
+            return
+
+        if language is None:
+            language = get_application_email_language(application=application)
+
+        email_type = EmailType.APPLICATION_RECEIVED
+        context = get_context_for_application_received(language=language)
+        send_emails_in_batches_task.delay(
+            recipients=recipients,
+            subject=context["title"],
+            text_content=render_text(email_type=email_type, context=context),
+            html_content=render_html(email_type=email_type, context=context),
+        )
+
+    @staticmethod
+    def send_application_in_allocation_emails() -> None:
+        """Sends emails to applicants when application round has entered the allocation phase."""
+        applications = Application.objects.should_send_in_allocation_email().order_by("created_date")
+        if not applications:
+            # TODO: Should we log this?
```

Comment:
If this is run as a periodic task, then the count will be zero most of the time.

Reply:
Removed.